modules/broker-auth/src/main/java/org/wso2/broker/auth/authentication/authenticator/impl/JaasAuthenticator.java

```diff
@@ -0,0 +1,93 @@
+/*
+ *   Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *   WSO2 Inc. licenses this file to you under the Apache License,
+ *   Version 2.0 (the "License"); you may not use this file except
+ *   in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing,
+ *   software distributed under the License is distributed on an
+ *   "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *   KIND, either express or implied.  See the License for the
+ *   specific language governing permissions and limitations
+ *   under the License.
+ *
+ */
+package org.wso2.broker.auth.authentication.authenticator.impl;
+
+import org.wso2.broker.auth.BrokerAuthConfiguration;
+import org.wso2.broker.auth.BrokerAuthConstants;
+import org.wso2.broker.auth.BrokerAuthException;
+import org.wso2.broker.auth.authentication.authenticator.Authenticator;
+import org.wso2.broker.auth.authentication.sasl.plain.PlainSaslCallbackHandler;
+import org.wso2.broker.auth.user.UserStoreManager;
+import org.wso2.broker.common.StartupContext;
+
+import java.util.Map;
+import javax.security.auth.login.AppConfigurationEntry;
+import javax.security.auth.login.Configuration;
+import javax.security.auth.login.LoginContext;
+import javax.security.auth.login.LoginException;
+
+/**
+ * An implementation of @{@link Authenticator} for provide authentication using
+ * Jaas @{@link javax.security.auth.spi.LoginModule}.
+ */
+public class JaasAuthenticator implements Authenticator {
+
+    @Override
+    public void initialize(StartupContext startupContext,
+                           BrokerAuthConfiguration.AuthenticationConfiguration authenticationConfiguration) {
+        UserStoreManager userStoreManager = startupContext.getService(UserStoreManager.class);
```

Comment:
We need to externalize the UserStoreManager class as well. Maybe in a later release. Currently, this is done in Main class. We need to move the initialization of UserStoreManager out of Main class.

Reply:
I have created a https://github.com/wso2/message-broker/issues/232 to track this.